browser/ui/tabs/shared_pinned_tab_service.cc

```diff
@@ -121,11 +122,14 @@ SharedPinnedTabService::GetTabRendererDataForDummyContents(
       DummyContentsData::FromWebContents(maybe_dummy_contents);
   DCHECK(dummy_contents_data);
 
-  const auto& pinned_tab_data = pinned_tab_data_.at(index);
-  DCHECK_EQ(dummy_contents_data->shared_contents(),
-            pinned_tab_data.shared_contents);
+  for (auto& pinned_tab_data : pinned_tab_data_) {
+    if (pinned_tab_data.shared_contents ==
+        dummy_contents_data->shared_contents()) {
+      return &pinned_tab_data.renderer_data;
+    }
+  }
 
-  return &pinned_tab_data.renderer_data;
+  NOTREACHED_NORETURN();
```

Comment:
1. The out-of-range exception caused the test failure. I assume there's a timing that pinned_tab_data_ isn't updated yet during pinned tab synchronization